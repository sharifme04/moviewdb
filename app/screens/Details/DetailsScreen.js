import React, {useEffect} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {
  ScrollView,
  ActivityIndicator,
  Image,
  Button,
  View,
  SafeAreaView,
  Text,
} from 'react-native';
import Config from 'react-native-config';
import {detailsMovies} from '../../actions/movies';
import {addFavourite, removeFavourite} from '../../actions/favourite';
import {theme} from '../../theme/theme';
import styles from './Style';

const DetailsScreen = ({route}) => {
  const dispatch = useDispatch();
  const {id} = route?.params || '';
  const movie = useSelector((state) => state.movies?.movie);
  const favourite = useSelector((state) => state.favourite?.favouriteMovies);

  useEffect(() => {
    dispatch(detailsMovies(id));
  }, [id, dispatch]);

  return (
    <SafeAreaView style={styles.flexContiainer}>
      <View style={styles.flexContiainer}>
        <ScrollView>
          <View style={styles.body}>
            <View>
              {movie?.poster_path ? (
                <Image
                  style={styles.image}
                  source={{
                    uri: `${Config.IMAGE_BASE_PATH}/w500/${movie?.poster_path}`,
                  }}
                />
              ) : (
                <ActivityIndicator size="large" color="#0000ff" />
              )}
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>
                Movie Name: {movie?.title}
              </Text>
              <Text style={styles.sectionDescription}>
                Overview: {movie?.overview}
              </Text>
              <Text style={styles.sectionDescription}>
                Release date: {movie?.release_date}
              </Text>
              <Text style={styles.sectionDescription}>
                Status: {movie?.status}
              </Text>
              <Text style={styles.sectionDescription}>
                Popularity: {movie?.popularity}
              </Text>
              <Text style={styles.sectionDescription}>
                Homepage: {movie?.homepage}
              </Text>
              <View>
                <Text style={styles.sectionDescription}>
                  Spoken languages:{' '}
                </Text>
                {movie?.spoken_languages?.map((e, i) => (
                  <View key={i} style={styles.spokenLanguage}>
                    <Text style={styles.spokenText}>{e.name}</Text>
                  </View>
                ))}
              </View>
            </View>
          </View>
        </ScrollView>
        {favourite?.includes(movie?.title) ? (
          <Button
            title="Remove from favorites"
            color={theme.colors.buttonPrimary}
            onPress={() => dispatch(removeFavourite(movie?.title))}
          />
        ) : (
          <Button
            title="Add to favorites"
            color={theme.colors.green}
            onPress={() => dispatch(addFavourite(movie?.title))}
          />
        )}
      </View>
    </SafeAreaView>
  );
};

export default DetailsScreen;
