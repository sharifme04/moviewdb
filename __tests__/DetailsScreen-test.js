import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import DetailsScreen from '../app/screens/Details/DetailsScreen';
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
const store = mockStore({});

const details = {
  adult: false,
  backdrop_path: '/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg',
  belongs_to_collection: {
    id: 86311,
    name: 'The Avengers Collection',
    poster_path: '/tqXiOD5rTyHgabO73Tpw9JDbd88.jpg',
    backdrop_path: '/zuW6fOiusv4X9nnW3paHGfXcSll.jpg',
  },
  budget: 300000000,
  genres: [
    {id: 12, name: 'Adventure'},
    {id: 28, name: 'Action'},
    {id: 878, name: 'Science Fiction'},
  ],
  homepage: 'https://www.marvel.com/movies/avengers-infinity-war',
  id: 299536,
  imdb_id: 'tt4154756',
  original_language: 'en',
  original_title: 'Avengers: Infinity War',
  overview:
    'As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.',
  popularity: 134.549,
  poster_path: '/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg',
  production_companies: [],
  production_countries: [],
  release_date: '2018-04-25',
  revenue: 2046239637,
  runtime: 149,
  spoken_languages: [],
  status: 'Released',
  tagline: 'An entire universe. Once and for all.',
  title: 'Avengers: Infinity War',
  video: false,
  vote_average: 8.3,
  vote_count: 19430,
};

test('DetailsScreen', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailsScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('DetailsScreen with details data', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailsScreen movie={details} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('DetailsScreen with details empty object ', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailsScreen movie={{}} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
