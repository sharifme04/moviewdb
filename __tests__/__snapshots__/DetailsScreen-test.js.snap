// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#0000ff"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "600",
                }
              }
            >
              Movie Name: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Overview: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Release date: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Status: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Popularity: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Homepage: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginTop": 10,
                    "marginVertical": 1,
                    "padding": 5,
                  }
                }
              >
                Spoken languages:
                 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#00e64d",
              },
            ]
          }
        >
          Add to favorites
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`DetailsScreen with details data 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#0000ff"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "600",
                }
              }
            >
              Movie Name: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Overview: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Release date: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Status: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Popularity: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Homepage: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginTop": 10,
                    "marginVertical": 1,
                    "padding": 5,
                  }
                }
              >
                Spoken languages:
                 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#00e64d",
              },
            ]
          }
        >
          Add to favorites
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`DetailsScreen with details empty object  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 25,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#0000ff"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "600",
                }
              }
            >
              Movie Name: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Overview: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Release date: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Status: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Popularity: 
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 10,
                  "marginVertical": 1,
                  "padding": 5,
                }
              }
            >
              Homepage: 
            </Text>
            <View>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginTop": 10,
                    "marginVertical": 1,
                    "padding": 5,
                  }
                }
              >
                Spoken languages:
                 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#00e64d",
              },
            ]
          }
        >
          Add to favorites
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
